torchvision/prototype/features/_mask.py

```diff
@@ -9,13 +9,36 @@
 
 
 class Mask(_Feature):
+    @classmethod
+    def _wrap(cls, tensor: torch.Tensor) -> Mask:
+        return tensor.as_subclass(cls)
+
+    def __new__(
+        cls,
+        data: Any,
+        *,
+        dtype: Optional[torch.dtype] = None,
+        device: Optional[Union[torch.device, str, int]] = None,
+        requires_grad: bool = False,
+    ) -> Mask:
+        tensor = cls._to_tensor(data, dtype=dtype, device=device, requires_grad=requires_grad)
+        return cls._wrap(tensor)
+
+    @classmethod
+    def wrap_like(
+        cls,
+        other: Mask,
+        tensor: torch.Tensor,
+    ) -> Mask:
+        return cls._wrap(tensor)
```

Comment:
Why can't we just call `tensor.as_subclass(cls)` here ? I'm not sure to understand the point of introducing `_wrap` function.

Reply:
You are right, if you only look at `Mask` it makes little sense. However, not having it would diverge the implementation from the other features. In https://github.com/pytorch/vision/pull/6718#discussion_r989999849 I've shown that it basically has no runtime implications and thus I would favor consistency over micro optimization, i.e. a single method call.